Turn on exact_dtype by default on test_sparse.py (#34489)

terhuhf opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34542 Turn on exact_dtype by default on test_sparse.py (#34489)**

Turn on exact_dtype by default on test_sparse.py (#34489)

Summary:
Pull Request resolved: #34489

Test Plan:

```
python test/test_sparse.py
```

Reviewers:

Subscribers:

Tasks:
https://github.com/pytorch/pytorch/issues/34489

Tags:

Differential Revision: [D20369764](https://our.internmc.facebook.com/intern/diff/D20369764)

terhuhf committed:
<Replace this line with a title. Use 1 line only, 67 chars or less>
Turn on exact_dtype by default on test_sparse.py (#34489)

Summary:
Pull Request resolved: #34489

Test Plan:

```
python test/test_sparse.py
```

Reviewers:

Subscribers:

Tasks:
https://github.com/pytorch/pytorch/issues/34489

Tags:

[ghstack-poisoned]

terhuhf commented on test/test_sparse.py:
```diff
@@ -227,8 +228,11 @@ def test_tensor(x, res):
-            self.assertEqual(res, self.safeToDense(x))
+            # We dont have to_dense for half types, so we don't request 
+            # exact_dtype when testing. We allow to_dense to convert dtype
+            # to torch.float64
```
What is the semantics of to_dense working with torch.float32?

ezyang replied:
Should work just fine.

albanD commented on test/test_sparse.py:
```diff
@@ -227,8 +228,11 @@ def test_tensor(x, res):
             x.to_dense()
-            self.assertEqual(res, x.to_dense())
-            self.assertEqual(res, self.safeToDense(x))
+            # We dont have to_dense for half types, so we don't request 
```
if the problem is only with half, why not use `exact_dtype=x.dtype is not torch.float16` ?

ezyang commented on test/test_sparse.py:
```diff
@@ -243,6 +247,7 @@ def fn(x):
         # we don't have to_dense for half types on CPU because it is implemented
         # with a slower add_ operation
         for dtype in [torch.float16, torch.float64] if self.device != 'cpu' else [torch.float64]:
+
```
nit: Avoid extraneous whitespace changes when working on diffs! They make reviewers unhappy ;)

ezyang reviewed (approved):
Looks great!

terhuhf committed:
Update on "Turn on exact_dtype by default on test_sparse.py (#34489)"


Turn on exact_dtype by default on test_sparse.py (#34489)

Summary:
Pull Request resolved: #34489

Test Plan:

```
python test/test_sparse.py
```

Reviewers:

Subscribers:

Tasks:
https://github.com/pytorch/pytorch/issues/34489

Tags:

[ghstack-poisoned]
